cmd/ctr-layertool/commands/decryptlayerstream/decryptlayerstream.go

```diff
@@ -0,0 +1,133 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package decryptlayerstream
+
+import (
+	"bufio"
+	"io"
+	"os"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cmd/ctr-layertool/commands/utils"
+	"github.com/containerd/containerd/images/encryption"
+	"github.com/containerd/typeurl"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+func init() {
+	typeurl.Register(&containerd.LayerToolDecryptData{}, "LayerToolDecryptData")
+}
+
+// Command is the cli command for managing images
+var Command = cli.Command{
+	Name:    "decrypt-layer-stream",
+	Aliases: []string{"dls"},
+	Usage:   "decrypt a layer stream",
+	Flags: []cli.Flag{
+		cli.IntFlag{
```

Comment:
I don't think we need flags for this, these will be consistent fds

Reply:
I removed them.